tests/decoder_tests.rs

```diff
@@ -356,6 +356,97 @@ fn gainmap_oriented() {
     assert_eq!(decoder.gainmap().image.imir_axis, None);
 }
 
+// From avifgainmaptest.cc
+#[test]
+fn decode_unsupported_version() {
+    // The two test files should produce the same results:
+    // One has an unsupported 'version' field, the other an unsupported
+    // 'minimum_version' field, but the behavior of these two fiels is the same.
+    for image in [
+        "unsupported_gainmap_version.avif",
+        "unsupported_gainmap_minimum_version.avif",
+    ] {
+        // Parse with various enable_decoding_gainmap and enable_parsing_gainmap_metadata
+        // settings.
+        let mut decoder = get_decoder(image);
+        decoder.settings.enable_decoding_gainmap = false;
+        decoder.settings.enable_parsing_gainmap_metadata = false;
+        let mut res = decoder.parse();
+        assert!(res.is_ok());
+        // Gainmap found.
+        assert_eq!(decoder.gainmap_present(), true);
```

Comment:
this can simply be `assert!(decoder.gainmap_present());` since we are asserting a boolean. ditto for all the boolean asserts below.